[master] Update chocolatey.py / Search for choco.exe

TiHiNi opened the pull request:
## chocolatey.exe was renamed to choco.exe by Chocolatey in Version 2.0.0


### What does this PR do?
Add the search for choco.exe to confirm that Chocolatey is installed.


### What issues does this PR fix or reference?
Fixes: #64427 

### Previous Behavior
The module only searched for chocolatey.exe

### New Behavior
The module now search for choco.exe, too.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

TiHiNi committed:
Update chocolatey.py / Search for choco.exe

chocolatey.exe was renamed to choco.exe by Chocolatey in Version 2.0.0

Ch3LL reviewed (changes requested):
This will require a changelog  and test coverage. This would probably be the best place for test coverage: tests/pytests/unit/modules/test_chocolatey.py

TiHiNi committed:
Update chocolatey.py

Change environ key of new lication to "ProgramDataChoco" to not overwrite the environ of the chocolatey.exe path.

TiHiNi committed:
Update test_chocolatey.py

Add test for choco.exe in ProgramData

TiHiNi committed:
Update chocolatey.py

Bug Fix the last commit

TiHiNi committed:
Add changelog

TimHiCe commented:
@Ch3LL thanks for your info. I saw it, too. I'm very new to GitHub and specially Python. I hoped to solve the problem with a little change. Changing chocolatey.py was easy, but I struggled with the unit-test. So, sorry for some extra circles.

I downloaded a local copy of the repo. Now I'm setting up the test env to run the test locally and commit a better unit-test shortly. 

You can safely wait with approval until next commit.

Ch3LL commented:
@TimHiCe just following up here. Any direction I can help with?

TimHiCe commented:
@Ch3LL thanks for your question. I'm working on to get the test suit locally running. That was a lot trial and error but meanwhile I'm getting closer to the goal.  I think I can come back the next days with a correct unit test or specific questions.

twangboy committed:
Fix pre-commit

twangboy committed:
Fix failing tests

TimHiCe commented:
Sorry for my late reply – I was some days on vacation.

@twangboy and @s0undt3ch Thank you for fixing the tests.
I spend a lot of hours to get the tests locally running but until now it doesn’t. But that’s a topic for another issue.

@Ch3LL Since I’m very new here, do I have to do more things on this pull request? Or it is OK how it is now?

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->

s0undt3ch commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[<img src="https://img.shields.io/github/pulls/detail/state/saltstack/salt/64672">](https://github.com/saltstack/salt/pull/64672)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->
